client/src/pages/TaskFormPage/TaskFormPage.js

```diff
@@ -1,9 +1,80 @@
-import React from "react";
+import React, { useState } from "react";
 import styles from "./TaskFormPage.module.css";
 import { InputBox } from "../../components/InputBox/InputBox";
 import { CredentialsButton } from "../../components/CredentialsButton/CredentialsButton.js";
+import { DateSelector } from "../../components/DateSelector/DateSelector.js";
+import { instance } from "../../axios";
 
 const TaskFormPage = () => {
+  const getISOString = (date) => {
+    return date.toISOString().slice(0, -5);
+  };
+
+  const [errorMessage, setErrorMessage] = useState("");
+  const [taskFormData, setTaskFormData] = useState({
+    title: "",
+    user_id: "620710de5b72a4271a59eabe", // TODO : figure out where to get it from
+    location: "",
+    description: "",
+    startDate: getISOString(new Date()),
+    endDate: getISOString(new Date()),
+    isStarted: false,
+  });
+
+  const updateTaskFormData = (e, value) => {
+    const newTaskFormData = { ...taskFormData };
+    newTaskFormData[value] = e.target.value;
+    setTaskFormData(newTaskFormData);
+  };
+
+  const updateTaskDate = (newDate, value) => {
+    const newTaskFormData = { ...taskFormData };
+    newTaskFormData[value] = getISOString(newDate);
+    setTaskFormData(newTaskFormData);
+  };
+
+  const validateTaskFormData = () => {
+    if (!taskFormData.title || taskFormData.title.length > 500) {
+      return "Invalid Task Title (max: 500 characters)";
+    }
+
+    if (!taskFormData.location || taskFormData.location.length > 500) {
+      return "Invalid Task Location (max: 500 characters)";
+    }
+
+    if (!taskFormData.description || taskFormData.description.length > 500) {
+      return "Invalid Task Description (max: 500 characters)";
+    }
+
+    const startDate = new Date(taskFormData.startDate);
+    const endDate = new Date(taskFormData.endDate);
+    if (startDate.getTime() > endDate.getTime()) {
+      return "Start date must come before end date";
+    }
+
+    return ""; // empty string means successful validation
+  };
+
+  const createTaskHandler = async () => {
+    setErrorMessage("");
+    const validationResult = validateTaskFormData();
+    console.log(validationResult);
+    if (!validationResult) {
```

Comment:
I would change this to `if (validationResult !== "")` just because the error message doesn't seem to display (only on console)
![image](https://user-images.githubusercontent.com/24628243/163062216-04924ec1-1deb-4ede-bc9c-d85a2ee921e6.png)

Reply:
The error message display has not been implemented (see PR checklist and standup). I've just set up a part of the error logic at the moment.

Reply:
Noted!